### DISCUSSION TITLE
Calculate coverage after track list was loaded

### DISCUSSION THREAD

**Discussion by TheIcyStar** [MKT-Playbook/Hub#16] [Feature requests]
Hello! Whenever I compare several d/k/gs I usually open a lot of coverage lists. However, every time it gets opened, there's a substantial delay before all of the favorite courses are visible. Of course, this is probably due to coverage calculation. But could that be done after the list is rendered without calculation, similar to the favored course list?

Basically, first, the list of courses gets rendered, then one by one the coverage-specific information gets added afterward. This way, you could see the full list of courses right away and not have to deal with an awkward scroll.

Thanks, and big ❤️ for making this awesome app!

**Comment by TitusPosiadly** [comment]
Hey there, thanks for the feedback!

Yes, I noticed quite a bit of performance issues a while ago and tried to research ways to fix this for the last couple of months whenever I had the time. I think I now know some good changes I can implement to get around this and it will likely be the next thing I will be working on. I am just currently struggling to find the time to work on the app, but I hope I can get some things done next week or so.

Thanks again for letting me know and for the kind words ❤️
